server/task/task-memory/src/main/scala/org/apache/james/task/eventsourcing/DecisionProjection.scala

```diff
@@ -32,7 +32,7 @@ case class DecisionProjection(status: Status, latestUpdateAdditionalInformationU
         case event: Cancelled => DecisionProjection(Status.CANCELLED, event.additionalInformation.map(_.timestamp))
         case event: Completed => DecisionProjection(Status.COMPLETED, event.additionalInformation.map(_.timestamp))
         case event: Failed => DecisionProjection(Status.FAILED, event.additionalInformation.map(_.timestamp))
-        case event: AdditionalInformationUpdated => DecisionProjection(status, Some(event.additionalInformation.timestamp))
+        case event: AdditionalInformationUpdated => DecisionProjection(Status.IN_PROGRESS, Some(event.additionalInformation.timestamp))
```

Comment:
I think I get the change, but what's the point of keeping status as an argument to the method update then?

Reply:
> I think I get the change, but what's the point of keeping status as an argument to the method update then?

Status is not an argument of the update method but part of the state of DecisionProjection and as such we need it.

This change is that the event fully determines the Decision projection disregardless of previous state (prerequisite for snapshot) but do not change the fact that status is needed to take a decision.

Reply:
Case class indeed, my bad